src/Gui/ViewProviderGroupExtension.cpp

```diff
@@ -171,24 +218,56 @@ void ViewProviderGroupExtension::extensionHide() {
     ViewProviderExtension::extensionHide();
 }
 
-bool ViewProviderGroupExtension::extensionOnDelete(const std::vector< std::string >& ) {
+bool ViewProviderGroupExtension::extensionOnDelete(const std::vector< std::string >& subNames) {
 
     auto* group = getExtendedViewProvider()->getObject()->getExtensionByType<App::GroupExtension>();
-    // If the group is nonempty ask the user if they want to delete its content
-    if (group->Group.getSize() > 0) {
-        QMessageBox::StandardButton choice =
-            QMessageBox::question(getMainWindow(), QObject::tr ( "Delete group content?" ),
-                QObject::tr ( "The %1 is not empty, delete its content as well?")
-                    .arg ( QString::fromUtf8 ( getExtendedViewProvider()->getObject()->Label.getValue () ) ),
-                QMessageBox::Yes | QMessageBox::No, QMessageBox::Yes );
-
-        if (choice == QMessageBox::Yes) {
-            Gui::Command::doCommand(Gui::Command::Doc,
-                    "App.getDocument(\"%s\").getObject(\"%s\").removeObjectsFromDocument()"
-                    , getExtendedViewProvider()->getObject()->getDocument()->getName()
-                    , getExtendedViewProvider()->getObject()->getNameInDocument());
-        }
+    
+    std::vector<App::DocumentObject*> directChildren = group->Group.getValues();
+    
+    // just delete without messagebox if group is empty
+    if (directChildren.empty()) {
+        return true;
+    }
+    
+    std::vector<App::DocumentObject*> allDescendants;
+    if (getExtendedViewProvider()->getObject()->isDerivedFrom<App::DocumentObjectGroup>()) {
+        auto* docGroup = static_cast<App::DocumentObjectGroup*>(getExtendedViewProvider()->getObject());
+        allDescendants = docGroup->getAllChildren();
+    } else {
+        allDescendants = directChildren;
+    }
```

Comment:
How about the following instead?
```suggestion
    const auto* docGroup = freecad_cast<App::DocumentObjectGroup*>(getExtendedViewProvider()->getObject());
    auto* allDescendants = docGroup ? docGroup->getAllChildren() : directChildren;
```